include/igl/copyleft/cgal/SelfIntersectMesh.h

```diff
@@ -635,14 +635,32 @@ inline bool igl::copyleft::cgal::SelfIntersectMesh<
   const std::vector<std::pair<Index,Index> > shared)
 {
   using namespace std;
+  
+  auto opposite_vertex = [](const Index a0, const Index a1) {
```

Comment:
Please do not mix up contributions in this PR. Could you keep the contrib to separate branches so we can review them separately? Thanks!

Reply:
It wasn't supposed to be included. I pushed this change after I created the pull request.

Reply:
I've reverted the commit.

Reply:
looks cool though. maybe we should test and merge this as a separate PR?

Reply:
Ok, I'll cancel this one and create a new one.

Reply:
@alecjacobson this was taken from #1413. I've asked @qnzhou to review and approve the PR, but it's been sitting in limbo since >1y ...